modules/http-server/src/test/java/org/glassfish/grizzly/http/server/NetworkListenerTest.java

```diff
@@ -241,7 +243,14 @@ public void testGracefulShutdownGracePeriodExpired() throws Exception {
             Future<HttpServer> gracefulFuture = server.shutdown(1, TimeUnit.SECONDS);
 
             final BufferedReader reader = new BufferedReader(new InputStreamReader(c.getInputStream(), Charsets.UTF8_CHARSET));
-            final String content = reader.readLine();
+            String content;
+            try {
+                // null is expected here as the grace period will expire
```

Comment:
This comment doesn't make much sense. Is `null` expected on this line? Wouldn't it result in a nullptr exception?
Or it's expected that reader.readLine() throws IOException and then it's OK to set the content to `null`?

I think what you want to say is that `content == null` means the grace period expired. And on newer JDKs the behavior changes - an exception is thrown if grace period expires, instead of returning null.

What about this comment:

```suggestion
                // this should result in null in older JDKs or IOException on newer JDKs as the grace period expires 
```

Reply:
> Perhaps I should add a little more explanation. I understand why you were confused. I checked the actual JDK source and noticed that the behavior changed (JDK17 vs. JDK25). 

Maybe this https://bugs.openjdk.org/browse/JDK-8335135 - https://github.com/openjdk/jdk/commit/b6ffb442acb4a222f017868433eff213d9b84ed8

Reply:
Thanks @carryel, makes perfect sense now! I was maybe a little too pedantic. i just thought it’s good to leave a clear comment in the code for people that come across it in the future :)